pints/_transformation.py

```diff
@@ -0,0 +1,574 @@
+        Returns a transformed log-PDF class.
+        """
+        return TransformedLogPDF(log_pdf, self)
+
+    def apply_error_measure(self, error_measure):
+        """
+        Returns a transformed error measure class.
+        """
+        return TransformedErrorMeasure(error_measure, self)
+
+    def apply_boundaries(self, boundaries):
+        """
+        Returns a transformed boundaries class.
+        """
+        return TransformedBoundaries(boundaries, self)
+
+    def jacobian(self, x):
+        """
+        Returns the Jacobian for a parameter vector ``x`` in the search space.
+
+        *This is an optional method. It is needed when `sigma0` is provided in
+        :meth:`pints.OptimisationController` or :meth:`pints.MCMCController`,
+        or when the method requires ``evaluateS1()``.*
+        """
+        raise NotImplementedError
+
+    def log_jacobian_det(self, x):
+        """
+        Returns the logarithm of the absolute value of the Jacobian
+        determinant for a parameter vector ``x`` in the search space.
+
+        *This is an optional method. It is needed when transformation is
+        performed on :class:`LogPDF` and/or that requires ``evaluateS1()``;
+        e.g. not necessary if it's used for :class:`ErrorMeasure` without
+        :meth:`ErrorMeasure.evaluateS1()`.*
+        """
+        return np.log(np.abs(np.linalg.det(self.jacobian(x))))
+
+    def n_parameters(self):
+        """
+        Returns the dimension of the parameter space this transformation is
+        defined over.
+        """
+        raise NotImplementedError
+
+    def to_model(self, x):
+        """
+        Returns the inverse of transformation of a vector from the search space
+        ``x`` to the model parameter space ``p``.
+        """
+        raise NotImplementedError
+
+    def to_search(self, p):
+        """
+        Returns the forward transformation of a vector from the model parameter
+        space ``p`` to the search space ``x``.
+        """
+        raise NotImplementedError
+
+
+class TransformedLogPDF(pints.LogPDF):
+    """
+    A log-PDF that is transformed from the model space to the search space.
+
+    Extends :class:`pints.LogPDF`.
+
+    Parameters
+    ----------
+    log_pdf
+        A :class:`pints.LogPDF`.
+    transform
+        A :class:`pints.Transform`.
+    """
+    def __init__(self, log_pdf, transform):
+        self._log_pdf = log_pdf
+        self._transform = transform
+        self._n_parameters = self._log_pdf.n_parameters()
+        if self._transform.n_parameters() != self._n_parameters:
+            raise ValueError('Number of parameters for log_pdf and transform '
+                             'must match.')
+
+    def __call__(self, x):
+        # Get parameters in the model space
+        p = self._transform.to_model(x)
+
+        # Compute LogPDF in the model space
+        logpdf_nojac = self._log_pdf(p)
+
+        # Calculate the PDF change of variable. From Wikipedia:
```

Comment:
This kind of stuff goes in the main docstrings. Users will want to know! (Or will want to be assured that whoever wrote the class did some fancy maths so that they don't have to)

Reply:
Haha! OK, I thought this was quite boring, so didn't want to bother users in the main docstrings.

Reply:
I can imagine users having some vague memory that you need to do something special, and then being happy that this method does it.

Or alternatively users who know exactly what you do, being worried that the docs don't mention that this is done!

Reply:
Updated docstring.